src/plugins/lua/LuaEvent.cpp

```diff
@@ -43,40 +43,56 @@ LuaEventProcessor::~LuaEventProcessor()
 	m_eventDefinitions.clear();
 }
 
-void LuaEventProcessor::AddEvent(std::string_view name, std::string_view expression, const sol::function& function)
+bool LuaEventProcessor::AddEvent(std::string_view name, std::string_view expression, const sol::function& function)
 {
+	// the number of events will always be fairly small, and this is a manual operation.
+	// If this is deemed too slow, the event names can be memoized in a set of string_views.
+	auto it = std::find_if(m_eventDefinitions.begin(), m_eventDefinitions.end(),
+		[&name](const std::unique_ptr<LuaEvent>& event) { return event->GetName() == name; });
+
+	if (it != m_eventDefinitions.end())
+	{
+		LuaError("Cannot create event %.*s, it is already defined.", name.length(), name.data());
```

Comment:
does this stop the script?

Reply:
It does not. It just shows an error (this is the same as bind)